### ISSUE TITLE
[BUG] Session Fixation on JacksonHttpSessionOAuth2AuthorizedClientRepository

### ISSUE THREAD

**Issue by Fradantim** [Azure/azure-sdk-for-java#23510]
**Describe the bug**
Hello There!

Our company Veracode implementation is finding a [CWE-384 Session Fixation](http://cwe.mitre.org/data/definitions/384.html) weakness in JacksonHttpSessionOAuth2AuthorizedClientRepository.java

https://github.com/Azure/azure-sdk-for-java/blob/facbc09f1453ed40aaa1c59ababa37ba1094da14/sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/JacksonHttpSessionOAuth2AuthorizedClientRepository.java#L65

Can't tell if facbc09f1453ed40aaa1c59ababa37ba1094da14 is the exact commit of v3.7.0, but in this commit the class looks exaclty the same as the one shipped v3.7.0.

***Exception or Stack Trace***
Add the exception log and stack trace if available

**To Reproduce**
pom.xml extract
``` xml
(...)
<parent>
	<groupId>org.springframework.boot</groupId>
	<artifactId>spring-boot-starter-parent</artifactId>
	<version>2.5.3</version>
	<relativePath /> <!-- lookup parent from repository -->
</parent>
	
<properties>
	<java.version>1.8</java.version>
</properties>

<dependencyManagement>
	<dependencies>
		<dependency>
			<groupId>com.azure.spring</groupId>
			<artifactId>azure-spring-boot-bom</artifactId>
			<version>3.7.0</version>
			<type>pom</type>
			<scope>import</scope>
		</dependency>
	</dependencies>
</dependencyManagement>

<dependencies>
	<dependency>
		<groupId>org.springframework.boot</groupId>
		<artifactId>spring-boot-starter-web</artifactId>
	</dependency>
	<dependency>
		<groupId>org.springframework.boot</groupId>
		<artifactId>spring-boot-starter-security</artifactId>
	</dependency>
	<dependency>
		<groupId>org.springframework.boot</groupId>
		<artifactId>spring-boot-starter-oauth2-resource-server</artifactId>
	</dependency>
	<dependency>
		<groupId>org.springframework.boot</groupId>
		<artifactId>spring-boot-starter-oauth2-client</artifactId>
	</dependency>
	<dependency>
		<groupId>com.azure.spring</groupId>
		<artifactId>azure-spring-boot-starter-active-directory</artifactId>
	</dependency>
</dependencies>
(...)
```

A blank project with those dependencies is enough to trigger the CWE on Veracode static scan.

***Code Snippet***
Add the code snippet that causes the issue.

**Expected behavior**

**Screenshots**
Veracode Triage Flaws screenshot:
![image](https://user-images.githubusercontent.com/5752124/129109718-097e90bf-8035-4b65-acc2-cb9501213061.png)

**Setup (please complete the following information):**
 - OS: Windows 10 / RHEL 7.6
 - IDE: -
 - Library/Libraries: com.azure.spring:azure-spring-boot-starter-active-directory:3.7.0
 - Java version: 8
 - App Server/Environment: Tomcat
 - Frameworks: Spring Boot

If you suspect a dependency version mismatch (e.g. you see `NoClassDefFoundError`, `NoSuchMethodError` or similar), please provide
- dependency tree (`mvn dependency:tree -Dverbose`)
  
**Additional context**

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [ x ] Bug Description Added
- [ x ] Repro Steps Added
- [ x ] Setup information Added

**Comment by chenrujun** [comment]
@Fradantim.
Thanks for reaching out. 
This problem will be fixed in version 4.0.0. 
My current plan in to release 4.0.0-beta.1 before 2021-11-01.

**Comment by chenrujun** [comment]
Closing this issue, because we decide to deprecate azure-spring-boot-starter-active-directory, and write samples to demonstrate how to use Azure Active Directory in Spring Boot application
